lib/gollum-lib/macro/video.rb

```diff
@@ -0,0 +1,17 @@
+module Gollum
+  class Macro
+    class Video < Gollum::Macro
+      def render (fname)
+        begin 
```

Comment:
> detects the video's MIME type so users don't need to specify it explicitly in the macro.

Thanks for the explanation, I understand the need now. What about the following approach then?
* Change the macro to accept two arguments: `fname` and `mime` (optional)
* If the repo is bare (`@wiki.repo.is_bare?`, I think), and `mime` is not specified, throw an error.
* If the repo is not bare, and `mime` is not specified, check the mime type in the currently implemented way.
* If the repo is not bare, and `mime` is specified, use `mime` as mime type.

Technically it would be possible to do the mime type check on bare repositories as well, using the `gollum-lib` API as I explained above. But I think it would be better not to do this for performance reasons (for large videos, this would be quite an expensive operation).